lib/skate/detours/trip_modification_publisher.ex

```diff
@@ -0,0 +1,108 @@
+defmodule Skate.Detours.TripModificationPublisher do
+  @moduledoc """
+    Connects to the MQTT Broker, then tracks and sends Trip Modifications to the Broker
+
+    References: https://github.com/mbta/ride_along/blob/fb440fb15dce22921fc4a141a125f3645da98b26/lib/ride_along/sql_publisher.ex
+  """
+  use GenServer
+
+  @default_name __MODULE__
+  def start_link(opts) do
+    if opts[:start] do
+      name = Keyword.get(opts, :name, @default_name)
+      GenServer.start_link(__MODULE__, opts, name: name)
+    else
+      :ignore
+    end
+  end
+
+  def publish_modification(
+        %Realtime.TripModification{} = modification,
+        opts
+      ) do
+    is_draft? = Keyword.get(opts, :is_draft?, false)
+    server = Keyword.get(opts, :server, @default_name)
+
+    # check that server exists?
```

Comment:
Does this comment mean that you were considering adding in a check here, but haven't yet?

Reply:
I left that for myself to consider later but never did 😅 
So I'm real unsure what to do here. 

If this module is non-critical, then we could theoretically use `GenServer.cast`, because we don't need any response, maybe the server is there, maybe it's not, does the caller want to know that?
That's what I was originally thinking when I did this.

Then, I was having issues with tests and not sure that the call was ever completing successfully, due to the "fire and forget" nature of `cast`.
So I changed it to `call` and then added the `id` return value to the return.

I never did go back and reevaluate fully if `call` _is_ what we want, and I'm not _certain_ it is. If Mqtt is optional, then a caller shouldn't care if it succeeds or not, and we shouldn't crash on any pattern matching of `{:ok, id} = publish_modification([...])`, at the same time, maybe callers _do_ want to know that, and we can selectively ignore the return value, but then that's more places that could crash.

If we use `call`, then we don't need to check the server is alive, but if we use `cast`, does the caller want to know if we know it didn't succeed because the process doesn't exist?(I'm not how to check for existence of a process either, but I'd _imagine_ it's easy)

Reply:
I think we care whether a message was sent, since if we fail to deliver a message, we'll need to re-send it. Otherwise some detour info won't make it to Transit Data, which wouldn't be great. So I think that `call` is the right 😎 _call_ 😎  here.

We should address a failed MQTT connection at some point, but a) I don't think we need to block this work on that, and b) I don't think we're in a good state to do that yet (I'm thinking that the _right_ handling is some kind of DB-backed retry, and we're not really set up for doing something in the DB yet).

For now, probably crashing via a bad pattern match is fine? Or the production code could just ignore the return value?

Reply:
> Or the production code could just ignore the return value?

Right, what I'm trying to get at is we'll need to _remember_ to do so, and to fail to remember to do something is only human :)

Reply:
True, but we have tools for that :D

My instinct is to not block merging this, and instead create a follow-up ticket that blocks the ["🤝 Detours that Skate produces are complete and accurate" milestone](https://app.asana.com/0/1206070633318963/1207635746210201/f).